internal/pkg/sysinfo/probes/linux/kernel_test.go

```diff
@@ -118,7 +118,14 @@ func TestKConfigProber(t *testing.T) {
 
 		option, err := probeKConfig(ensureKConfig("I_CERTAINLY_DONT_EXIST"))
 		assert.Equal(t, option, kConfigUnknown)
-		assert.NoError(t, err)
+		switch err := err.(type) {
```

Comment:
Can you use `errors.Is()` here? Looks like you'd be able to remove a case with it.

Reply:
Oops, I meant `errors.As()`.  Something like this ..

```
var nce *noKConfigsFound
if errors.As(err, &nce) {
  t.Log("system doesn't ....")
} else {
  assert.NoError(t, err)
}
```

Reply:
I've chosen a third option :-)